schema.json

```diff
@@ -0,0 +1,404 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "allOf": [{ "$ref": "#/$defs/PgRollMigration" }],
```

Comment:
What's the purpose of `allOf` here? The same types are still generated successfully if I remove this line.

Reply:
The `PgRollMigration` type actually duplicates a type we define here:

https://github.com/xataio/pgroll/blob/main/pkg/migrations/migrations.go#L46-L53

So it could probably be removed entirely.

Reply:
I missed those. What about I create: `Migration`, `Operation` and `Operations` and keep the `allOf` with `$ref` top level?

I can get it done tomorrow so it's ready for a final review.

Reply:
After checking today, it might not be possible to generate these and re-use them.

1. The `Operation` in pgroll it's not just an struct, it's an interface with methods.
2. The library hasn't added support for `anyOf` yet, they have a [PR open](https://github.com/omissis/go-jsonschema/pull/99) but stale

I think it's fine as-is leaving the top level `allOf: { $ref }` to reference the definition of `PgRollMigration` that doesn't clash with the internal `Migration` one.

I've subscribed to that PR, once they merge support for `anyOf` maybe we can accomodate both and fully re-use `Migration`, `Operations` and `Operation`.

WDYT?

Reply:
Thats fine from my POV. The main benefit we get from this is being able to generate the types for the different operations. 

Is this PR good to go from your side now?

Reply:
Yes, can be merged when you're good with it